prompt_toolkit/shortcuts/progress_bar/formatters.py

```diff
@@ -198,6 +199,29 @@ def get_width(self, progress_bar: 'ProgressBar') -> AnyDimension:
         return D.exact(max(all_lengths) * 2 + 1)
 
 
+class Counter(Formatter):
```

Comment:
In order to avoid duplication, can you create a similar ``Formatter`` for ``Total``, and then rewrite the ``Progress`` formatter so that it uses the other two underneath?

There is a ``merge_formatted_text`` utility to concatenate ``HTML`` objects.

Reply:
done